protocol_rest.go

```diff
@@ -140,7 +140,11 @@ func (r restClientProtocol) requestNeedsPrep(op *operation) bool {
 	return len(op.restTarget.vars) != 0 ||
 		len(op.request.URL.Query()) != 0 ||
 		op.restTarget.requestBodyFields != nil ||
```

Comment:
Need to drop this line `op.restTarget.requestBodyFields != nil ||`. It's covered by below.